server/controllers/conversation.js

```diff
@@ -0,0 +1,62 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+
+const Conversation = require('../models/Conversation');
+const User = require('../models/User');
+const Profile = require('../models/Profile');
+const Message = require('../models/Message');
+
+// @route GET/conversations
+//Get all conversations for a user
+exports.getConversations = asyncHandler(async (req,res,next) => {
+  try {
+    const user = await User.findById(req.user.id);
+    Conversation.find(
+      {
+      participants: { $all: [user.profile] },
+      },
+      async (err, conversations) => {
+        if (err) console.log (err);
+        const conversationData = await Promise.all(
+          conversations.map(async conversation => {
+            return {
```

Comment:
Also add lastMessage and seen status when returning conversation to be seen on frontend

Reply:
Refactored conversation controller, for increased readability and added last message and seen status as well.